Filter and Remove

rctauber opened the pull request:
See #272 

If this looks good, the only thing I have left to do is add in documentation for the site!

rctauber commented:
I need to update this command in the integration tests as well:
```
robot materialize --reasoner ELK --input emr_example.obo --term BFO:0000050 \
    filter -t BFO:0000050 \
    reduce --output results/emr_reduced.obo
```

I didn't see this before, and since we changed the behavior of filter this is now wrong. I'll fix this as I update the docs for `remove` and `filter`.

jamesaoverton commented on docs/remove.md:
```diff
@@ -1,23 +1,120 @@
+1. `CURIE=CURIE`
+2. `CURIE=<IRI>`
+3. `CURIE='literal'`
+4. `CURIE='literal'^^datatype`*
```
It's more work, but we shouldn't limit the available datatypes.

jamesaoverton commented:
We should not drop `FilterOperation.java`, even if we don't use it anymore. We should just point to the new approach in the docs for that file. And we might as well keep its test file.

How much does the new Guava dependency buy us?

A few examples still use `--entity`. They should use `--term`.

The options are now called `--term` and `--term-file` but we say 'entities' a lot in the documentation. I'm not sure what to do about that. We can leave it for now.

This is a major feature. I think it's good, but I want feedback from other developers and users.

rctauber commented:
> We should not drop FilterOperation.java, even if we don't use it anymore. We should just point to the new approach in the docs for that file. And we might as well keep its test file.

Added those back in, with a note on `FilterOperation.java`

> How much does the new Guava dependency buy us?

This was leftover from earlier, I think... I removed it and everything still runs fine.

> A few examples still use --entity. They should use --term.

Fixed.

> The options are now called --term and --term-file but we say 'entities' a lot in the documentation. I'm not sure what to do about that. We can leave it for now.

I changed a few of them to make more sense.

cmungall commented:
@rctauber can you resolve the conflicts in the imports, thanks!
